esphome/dashboard/dashboard.py

```diff
@@ -695,20 +695,40 @@ def get(self):
 
 class BoardsRequestHandler(BaseHandler):
     @authenticated
-    def get(self):
+    def get(self, platform: str):
         from esphome.components.esp32.boards import BOARDS as ESP32_BOARDS
         from esphome.components.esp8266.boards import BOARDS as ESP8266_BOARDS
         from esphome.components.rp2040.boards import BOARDS as RP2040_BOARDS
 
-        boards = {
-            "esp32": {key: val[const.KEY_NAME] for key, val in ESP32_BOARDS.items()},
-            "esp8266": {
-                key: val[const.KEY_NAME] for key, val in ESP8266_BOARDS.items()
-            },
-            "rp2040": {key: val[const.KEY_NAME] for key, val in RP2040_BOARDS.items()},
+        platform_to_boards = {
+            "esp32": ESP32_BOARDS,
+            "esp32s2": ESP32_BOARDS,
+            "esp32s3": ESP32_BOARDS,
+            "esp32c3": ESP32_BOARDS,
+            "esp8266": ESP8266_BOARDS,
+            "rp2040": RP2040_BOARDS,
+        }
+        # filter all ESP32 variants by requested platform
+        for variant, boards in platform_to_boards.items():
+            if not variant.startswith("esp32"):
+                continue
```

Comment:
We should only do this for the variant that's chosen. 
```suggestion
        if platform.startswith("esp32"):
            boards = platform_to_boards[variant]
```